pom.xml

```diff
@@ -211,7 +211,7 @@
       <dependency>
         <groupId>org.apache.httpcomponents</groupId>
         <artifactId>httpclient</artifactId>
-        <version>4.5.3</version>
+        <version>4.5.9</version>
```

Comment:
Properly best to use
```
  <dependency>
    <groupId>org.jenkins-ci.plugins</groupId>
    <artifactId>apache-httpcomponents-client-4-api</artifactId>
    <version>4.5.10-2.0</version>
  </dependency>
```

Reply:
I am not sure if this should be in dependencies but when added to dependency management has some required upper bound errors:
```
Require upper bound dependencies error for org.apache.httpcomponents:httpclient:4.5.9 paths to dependency are:
+-io.jenkins.plugins:gitlab-branch-source:1.4.5-SNAPSHOT
  +-io.jenkins.plugins:gitlab-api:1.0.6
    +-org.gitlab4j:gitlab4j-api:4.14.20
      +-org.glassfish.jersey.connectors:jersey-apache-connector:2.30.1
        +-org.apache.httpcomponents:httpclient:4.5.9
and
+-io.jenkins.plugins:gitlab-branch-source:1.4.5-SNAPSHOT
  +-org.jenkins-ci.plugins:git:3.10.0
    +-org.jenkins-ci.plugins:git-client:2.7.7
      +-org.jenkins-ci.plugins:apache-httpcomponents-client-4-api:4.5.10-2.0 (managed) <-- org.jenkins-ci.plugins:apache-httpcomponents-client-4-api:4.5.3-2.0
        +-org.apache.httpcomponents:httpclient:4.5.10
and
+-io.jenkins.plugins:gitlab-branch-source:1.4.5-SNAPSHOT
  +-org.jenkins-ci.plugins:git:3.10.0
    +-org.jenkins-ci.plugins:git-client:2.7.7
      +-org.jenkins-ci.plugins:apache-httpcomponents-client-4-api:4.5.10-2.0 (managed) <-- org.jenkins-ci.plugins:apache-httpcomponents-client-4-api:4.5.3-2.0
        +-org.apache.httpcomponents:httpmime:4.5.10
          +-org.apache.httpcomponents:httpclient:4.5.10
and
+-io.jenkins.plugins:gitlab-branch-source:1.4.5-SNAPSHOT
  +-org.jenkins-ci.plugins:git:3.10.0
    +-org.jenkins-ci.plugins:git-client:2.7.7
      +-org.jenkins-ci.plugins:apache-httpcomponents-client-4-api:4.5.10-2.0 (managed) <-- org.jenkins-ci.plugins:apache-httpcomponents-client-4-api:4.5.3-2.0
        +-org.apache.httpcomponents:fluent-hc:4.5.10
          +-org.apache.httpcomponents:httpclient:4.5.10
and
+-io.jenkins.plugins:gitlab-branch-source:1.4.5-SNAPSHOT
  +-org.jenkins-ci.plugins:git:3.10.0
    +-org.jenkins-ci.plugins:git-client:2.7.7
      +-org.jenkins-ci.plugins:apache-httpcomponents-client-4-api:4.5.10-2.0 (managed) <-- org.jenkins-ci.plugins:apache-httpcomponents-client-4-api:4.5.3-2.0
        +-org.apache.httpcomponents:httpclient-cache:4.5.10
          +-org.apache.httpcomponents:httpclient:4.5.10
and
+-io.jenkins.plugins:gitlab-branch-source:1.4.5-SNAPSHOT
  +-org.jenkins-ci.plugins:git:3.10.0
    +-org.jenkins-ci.plugins:git-client:2.7.7
      +-org.jenkins-ci.plugins:apache-httpcomponents-client-4-api:4.5.10-2.0 (managed) <-- org.jenkins-ci.plugins:apache-httpcomponents-client-4-api:4.5.3-2.0
        +-org.apache.httpcomponents:httpasyncclient:4.1.4
          +-org.apache.httpcomponents:httpclient:4.5.6
```

Adding to dependency fixes this 98fe899

Reply:
@oleg-nenashev Can you comment on this issue? gitlab4j's pom has httpclient in dependency management https://github.com/eclipse-ee4j/jersey/blob/master/pom.xml#L1614 while git's pom has httpclient in dependencies https://github.com/jenkinsci/apache-httpcomponents-client-4-api-plugin/blob/master/pom.xml#L51. Why are their conflicts when `apache-httpcomponents-client-4-api` is added to dependencymanagement?

Reply:
```
"nearest definition" means that the version used will be the closest one to your project in the tree of dependencies. For example, if dependencies for A, B, and C are defined as A -> B -> C -> D 2.0 and A -> E -> D 1.0, then D 1.0 will be used when building A because the path from A to D through E is shorter. You could explicitly add a dependency to D 2.0 in A to force the use of D 2.0.
```
Maven tries to use the nearest definition so adding `apache-httpcomponents-client-4-api` to pom dependency directly creating a short dependency tree path and thus picked up but it is not a direct dependency of the project so is it right to add `apache-httpcomponents-client-4-api` to dependencies?

Reply:
You can also revert the change and inside the dependency management just bump the httpclient to `<version>4.5.10</version>` and the dependency conflict will go away however you run the risk of dependency load order issues of apache http client :)